lusidtools/cocoon/utilities.py

```diff
@@ -730,8 +730,34 @@ def parse_args(args: dict):
     return vars(ap.parse_args(args=args)), ap
 
 
+def scale_quote_of_type(df: pd.DataFrame, mapping: dict, file_type: str="quotes") -> pd.DataFrame:
+    """
+
+    :param string file_type: File type of data default = "quotes"
+    :param pd.DataFrame df: DataFrame containing data to be scaled
+    :param mapping: mapping configuration containing quote_scalar dictionary
+    :return:
+    """
+
+    price_col = mapping[file_type]["quote_scalar"]["price"]
+    type_col = mapping[file_type]["quote_scalar"]["type"]
+    type_code = mapping[file_type]["quote_scalar"]["type_code"]
+    scale_factor = mapping[file_type]["quote_scalar"]["scale_factor"]
+
+    df["Prices Corrected"] = None
+
+    for index, row in df.iterrows():
+        if row[type_col] == type_code:
+            corrected_price = row[price_col] * scale_factor
+        else:
+            corrected_price = row[price_col]
+        df.at[
+            index, "Prices Corrected"] = corrected_price
```

Comment:
This isn't really a "correction", perhaps rename to "__adjusted_quote"

If we're adding additional columns we might need to adopt a convention to avoid any potential column name clashes, I'd suggest prefixing with "__" (double underscore)

Reply:
Actually, we probably need to check if our adjusted column already exists, we have 2 options:
1) raise a ValueError
2) pass in the name for the column and use that

Reply:
- changed to `__adjusted_quote`
- ` df["__adjusted_quote"] = None` creates the few a few lines before. Is it really necessary to check it still exists?

Reply:
- changed to __adjusted_quote
- the `__adjusted_quote` column is created only a few lines before, why is it necessary to check this only a few lines later?